README.md

```diff
@@ -199,25 +198,20 @@ class StackTest {
 markers. This way we can have java comments: `//`, python comments: `#` or
 similar preceding our markers. **This means code can still compile!**
 
-There are some rules for prefixing to observe:
+### There are some rules for prefixing to observe:
 
-`Sanitizer`:
+Prefixes are valid for a single block, and are defined on the same line as the `REPOBEE-SANITIZER-START` marker of that block. All text found before the `START` marker (stripped fom whitespace) counts as the prefix.
 
-* Determines prefix as any text (including whitespace) that comes before `REPOBEE-SANITIZER-START`
-* Only determines prefix on a block-to-block basis, meaning that the prefix
-  selected at a `START` marker must be used until and including the next `END`
-  marker
-  * This means that all `Sanitizer` blocks can have individual
-    prefixes
-* Code between replace and end markers **MUST** also be prefixed if one is used
-* Prefixes inside `REPLACE-WITH` blocks are removed when sanitizing
+All lines inside a `REPLACE` block must have the prefix (if used), which is then stripped from the line. All whitespace is preserved when sanitizing and only the first occurrence of a prefix on each line is removed, allowing code comments inside the `REPLACE` block to be preserved.
+
+> Pro-Tip: There can be any amount of whitespace on any line using a prefix, see example 4. This allows easy alignment of code
```

Comment:
I'm not sure I agree this is a pro tip, it kind of seems like _the_ way to use the replace block.

Reply:
Yeah, let's just remove it, people will figure that out, we also show it in the example